app/src/modules/settings/routes/data-model/field-detail/components/interface.vue

```diff
@@ -117,7 +117,34 @@ export default defineComponent({
 		});
 
 		const selectedInterface = computed(() => {
-			return interfaces.value.find((inter: InterfaceConfig) => inter.id === state.fieldData.meta?.interface);
+			// get the default interface
+			const customInterface: InterfaceConfig | undefined = interfaces.value.find(
+				(inter: InterfaceConfig) => inter.id === state.fieldData.meta.interface
+			);
+
+			if (customInterface) {
+				if (customInterface.component.name === 'interface-dropdown') {
+					// update the interface-value to the specific one to only allow values of the correct type
```

Comment:
We can't do any logic based on hardcoded interface/component name. `interface-dropdown` is just one of many, and the logic needs to work for custom extensions too (which can have any name)

Reply:
Ya I wasn't sure what this block does, but now I presume it's to avoid us being able to key in valid values? (in relation to your last comment below as well)

so do we just change this block back into this? Drawing blank on the validation portion atm..

```js
                const selectedInterface = computed(() => {
			return interfaces.value.find((inter: InterfaceConfig) => inter.id === state.fieldData.meta?.interface);
		});
```

Reply:
I think it might've been intended to force the options of the dropdown interface to use the "datetime" interface when using a date type for example, however we can't do that in this part of the codebase